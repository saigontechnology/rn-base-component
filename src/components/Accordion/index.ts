export * from './Accordion'
export * from './AccordionItem'
export * from './ToggleAnimation'
