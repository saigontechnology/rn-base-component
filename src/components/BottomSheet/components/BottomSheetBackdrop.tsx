import React, {memo, useMemo} from 'react'
import {TapGestureHandler, TapGestureHandlerGestureEvent} from 'react-native-gesture-handler'
import Animated, {
  runOnJS,
  useAnimatedGestureHandler,
  useAnimatedStyle,
  withTiming,
} from 'react-native-reanimated'
import styled from 'styled-components'
import {useBottomSheet} from '../hooks'
import type {ITheme} from 'src/theme'
import type {StyleProp, ViewStyle} from 'react-native'

const Container = styled(Animated.View)<{theme: ITheme}>(({theme}) => ({
  position: 'absolute',
  top: 0,
  left: 0,
  right: 0,
  bottom: 0,
  backgroundColor: theme?.colors?.black,
  opacity: theme?.opacity?.partiallyOpaque,
}))

export interface BottomSheetBackdropProps {
  style?: StyleProp<ViewStyle>
  children?: React.ReactNode
}

const BottomSheetBackdropComponent: React.FC<BottomSheetBackdropProps> = ({style, children}) => {
  const {animatedIsVisible, close} = useBottomSheet()

  const animatedStyle = useAnimatedStyle(
    () => ({
      opacity: withTiming(animatedIsVisible.value ? 0.5 : 0),
    }),
    [],
  )

  const containerStyle = useMemo(() => [animatedStyle, style], [animatedStyle, style])

  const gestureHandler = useAnimatedGestureHandler<TapGestureHandlerGestureEvent>(
    {
      onFinish: () => {
        runOnJS(close)()
      },
    },
    [],
  )

  return (
    <TapGestureHandler onGestureEvent={gestureHandler}>
      <Container style={containerStyle}>{children}</Container>
    </TapGestureHandler>
  )
}

const BottomSheetBackdrop = memo(BottomSheetBackdropComponent)
BottomSheetBackdrop.displayName = 'BottomSheetBackdrop'

export default BottomSheetBackdrop
