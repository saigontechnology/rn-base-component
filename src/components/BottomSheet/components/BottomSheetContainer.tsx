import React, {memo, useMemo} from 'react'
import type {ViewProps} from 'react-native'
import {StyleProp, StyleSheet, ViewStyle} from 'react-native'
import {GestureHandlerRootView} from 'react-native-gesture-handler'
import Animated, {useAnimatedProps} from 'react-native-reanimated'
import {useBottomSheet} from '../hooks'

export interface BottomSheetContainerProps {
  style?: StyleProp<ViewStyle>
  children?: React.ReactNode
}

const BottomSheetContainerComponent: React.FC<BottomSheetContainerProps> = ({style, children}) => {
  const {animatedIsVisible} = useBottomSheet()

  const animatedProps = useAnimatedProps(
    () => ({
      pointerEvents: animatedIsVisible.value ? 'auto' : ('none' as ViewProps['pointerEvents']),
    }),
    [],
  )

  const containerStyle = useMemo<StyleProp<ViewStyle>>(
    () =>
      StyleSheet.flatten([
        {
          ...StyleSheet.absoluteFillObject,
          justifyContent: 'flex-end',
        },
        style,
      ]),
    [style],
  )

  return (
    <Animated.View style={containerStyle} animatedProps={animatedProps}>
      <GestureHandlerRootView style={containerStyle}>{children}</GestureHandlerRootView>
    </Animated.View>
  )
}

const BottomSheetContainer = memo(BottomSheetContainerComponent)
BottomSheetContainer.displayName = 'BottomSheetContainer'

export default BottomSheetContainer
