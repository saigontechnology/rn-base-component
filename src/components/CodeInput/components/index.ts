export {ErrorText} from './ErrorText'
export {HelperText} from './HelperText'
