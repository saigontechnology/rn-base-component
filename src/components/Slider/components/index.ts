export * from './Thumb'
export * from './TrackPoint'
export * from './Track'
