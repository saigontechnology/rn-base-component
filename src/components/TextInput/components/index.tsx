export * from './CustomIcon'
export * from './ErrorText'
