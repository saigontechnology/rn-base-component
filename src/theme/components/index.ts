import {
  ButtonTheme,
  ButtonOutlineTheme,
  ButtonPrimaryTheme,
  ButtonSecondaryTheme,
  ButtonTransparentTheme,
} from './Button'
import {CountDownTheme} from './Countdown/Countdown'
import {CheckboxTheme} from './Checkbox'
import {TextTheme} from './Text'

export default {
  Button: ButtonTheme,
  ButtonOutline: ButtonOutlineTheme,
  ButtonPrimary: ButtonPrimaryTheme,
  ButtonSecondary: ButtonSecondaryTheme,
  ButtonTransparent: ButtonTransparentTheme,
  Checkbox: CheckboxTheme,
  Text: TextTheme,
  CountDown: CountDownTheme,
}
