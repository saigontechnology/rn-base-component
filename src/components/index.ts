import Accordion from './Accordion/Accordion'
import RadioButton from './RadioButton/RadioButton'
import Button from './Button'

import Progress from './Progress/Progress'
import Checkbox from './Checkbox/Checkbox'
import CodeInput from './CodeInput/CodeInput'
import Slider from './Slider/Slider'
import Card from './Card/Card'
import TextInput from './TextInput/TextInput'

export {Button, CodeInput, Checkbox, Progress, Slider, RadioButton, Card, TextInput, Accordion}
export * from './Text/Text'

