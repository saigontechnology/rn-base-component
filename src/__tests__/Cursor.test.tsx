import React from 'react'
import {Cursor} from '../components'
import {render} from '@testing-library/react-native'

describe('Cursor', () => {
  it('renders correctly', () => {
    const {toJSON} = render(<Cursor />)
    expect(toJSON()).toMatchSnapshot()
  })
})
