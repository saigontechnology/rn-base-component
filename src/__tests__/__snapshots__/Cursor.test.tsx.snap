// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cursor Component Basic Rendering renders correctly with default props 1`] = `
<Text
  style={
    [
      {
        "opacity": 0,
      },
      undefined,
    ]
  }
>
  |
</Text>
`;
